### PR TITLE
test: Add APK and RSA sign test

### PR DESCRIPTION
**Pull request by blueworrybear**
To increase test coverage.

See #229

### CODE REVIEW

**Comment by erikgeiser** [comment]
Thanks, let me just copy my remark from issue #229 to discuss it here: 

Thanks for your contribution @blueworrybear. Sorry, but I don't really understand what the sign.go code is actually testing. RSAVerifySHA1Digest is actually not directly used by nfpm but exported to be used in tests in other packages. As far as I currently understand it, it is not really necessary to pull in all the dependencies to test different types of keys. I might be wrong, though.

File: internal/sign/rsa_test.go [hunk 1]
```diff
@@ -49,3 +60,51 @@ func TestInvalidHash(t *testing.T) {
...
+		{
+			Type:  "PUBLIC KEY",
+			Bytes: x509Bytes,
+		},
+	}
+
+	for _, pemKey := range pemKeys {
+		cwd, _ := os.Getwd()
+
+		keyFile, err := ioutil.TempFile(cwd, "*.pem")
```
Comment (erikgeiser) [inline internal/sign/rsa_test.go:98]: Maybe just create the temp file in the OS temp folder (`ioutil.TempFile("", "*.pem")`). This way the working tree is not polluted if the test is interrupted by CTRL+C in the wrong time.

**Comment by blueworrybear** [comment]
> Thanks, let me just copy my remark from issue #229 to discuss it here:
> 
> Thanks for your contribution @blueworrybear. Sorry, but I don't really understand what the sign.go code is actually testing. RSAVerifySHA1Digest is actually not directly used by nfpm but exported to be used in tests in other packages. As far as I currently understand it, it is not really necessary to pull in all the dependencies to test different types of keys. I might be wrong, though.

I see, I misunderstand the comment of `RSAVerifySHA1Digest`. Since coverage report counts in `RSAVerifySHA1Digest`, should I keep the tests or remove it?

File: internal/sign/rsa_test.go (hunk 1 above)
Comment (blueworrybear) [reply internal/sign/rsa_test.go]: fixed!

**Comment by erikgeiser** [comment]
The function was added to verify APK signatures which are not that well documented. As far as the documentation goes, only RSA keys are supported. So I don't know if ECDSA does actually even work with APKs. In my very limited understanding of the crypto involved, I think ECDSA is also one of the worse elliptic curve schemes. If I'm correct the ECDSA key test case tests a scenario that does not happen in practice.

What I don't understand is that the ECDSA works at all given that I did a type assertion that should require the key to be an RSA key:

https://github.com/goreleaser/nfpm/blob/cfc81695484e824e305f83c83ff81e1bcc3a69c9/internal/sign/rsa.go#L98-L101

Testing the function even though it is not used in the actual production code is still not a bad idea but if you want you can remove the key creation and just test with the keys in the `testdata` directory that are used for the other tests.

**Comment by caarlos0** [comment]
> @caarlos0 wdyt?

sgtm

(sorry I'm late, notification got lost in a sea of notifications_

**Comment by erikgeiser** [comment]
I rebased this PR to `master` and replaced the RSA tests with detailed tests for different error conditions. It can be merged now if CI succeeds.

**Comment by caarlos0** [comment]
Awesome, thanks

**Pull request merged**